### PR TITLE
add vpn template

### PR DESCRIPTION
**Pull request by TheoBrigitte**
Towards: https://github.com/giantswarm/azure-operator/issues/311

This is an attempt to automates embedding of json files into Go source code.

A simpler approach would be to directly write a file like `virtual_network_gateway.json.go` but this makes manipulation of json tedious.

My motivation are:
- easier manipulation of json files (syntax highlighting, jq processing,  ...)
- keep the benefit of go constants

It requires to run `go generate` prior to `go build` in order to generate the go source file. This could be added as an new step in `architect` workflow.

### CODE REVIEW

File: service/controller/v5/resource/vpn/deployment.go [hunk 1]
```diff
@@ -1,5 +1,7 @@
 package vpn
 
+//go:generate go run github.com/logrusorgru/textFileToGoConst -in virtual_network_gateway.json -c vpn_gateway_json
```
Comment (kopiczko) [inline service/controller/v5/resource/vpn/deployment.go]: For the record. We talked with @TheoBrigitte. This would be fine (at least with me) but there are two problems.

* Easy to solve problem

We'd need to ensure we control the version of `textFileToGoConst` we run. We could solve that by just forking the repo to make sure.

* Hard to solve problem

In `main.json` we import templates from public URL. We could use go templating and create a big single template. The problem here is Azure is far from being perfect with reporting template errors. When we have them split into multiple files it's easier to find a problem in smaller scope while having a cryptic error message.

Bottom line:
We should find a solution which can be applied to the most complex case first. Which is `main.json` template.

**Comment by TheoBrigitte** [comment]
No progress here.
